### PR TITLE
(fix) Fix `Last completed date` in forms dashboard

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR fixes an issue where the `Last completed date` in the forms dashboard was not being displayed correctly. The logic that determines the last completed date was not correctly sorting the encounters by date, and was not using the most recent encounter date. This PR fixes these issues and ensures that the last completed date is the most recent encounter date for each form.

## Screenshots

https://github.com/user-attachments/assets/b0170024-f1fc-4f0f-86a3-a81a3cf1c97c

## Related Issue

https://openmrs.atlassian.net/browse/O3-4010

### CODE REVIEW

**Comment by denniskigen** [comment]
FYI @VeronicaMuthee @fanderson

File: packages/esm-patient-forms-app/src/types/index.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export interface ListResponse<T> {
 export interface CompletedFormInfo {
   form: Form;
   associatedEncounters: Array<EncounterWithFormRef>;
-  lastCompleted?: Date;
+  lastCompletedDate?: Date;
```
Comment (denniskigen) [inline packages/esm-patient-forms-app/src/types/index.ts:76]: I've renamed this value to `lastCompletedDate` for clarity.

File: packages/esm-patient-forms-app/src/hooks/use-forms.ts [hunk 2]
```diff
@@ -113,13 +114,15 @@ function mapToFormCompletedInfo(
 ): Array<CompletedFormInfo> {
   return allForms.map((form) => {
     const associatedEncounters = encounters.filter((encounter) => encounter.form?.uuid === form?.uuid);
-    const lastCompleted =
-      associatedEncounters.length > 0 ? new Date(associatedEncounters?.[0].encounterDatetime) : undefined;
```
Comment (denniskigen) [inline packages/esm-patient-forms-app/src/hooks/use-forms.ts:117]: This function determines the data of the most recent matching encounter by picking the `encounterDatetime` of the first encounter and creates a new Date object from it. If there are no associated encounters, it sets `lastCompleted` to undefined. 

This implementation is not accurate. The code assumes that `associatedEncounters` is already sorted with the most recent encounter first. However, there’s no guarantee of this sorting in the code. Second, it only checks the first encounter in the array `associatedEncounters?.[0]`, which may not be the most recent encounter if the array is not sorted. 

I've modified the code to make sure we’re getting the most recent date from `associatedEncounters`. We’d then create a new Date object from the most recent encounter's date, guaranteeing that that would be the last completed date.

**Comment by VeronicaMuthee** [comment]
Thanks for the fix @denniskigen and will retest once the PR is merged.

**Comment by ibacher** [comment]
This suggests to me two tickets for the backend:

1. When the list of encounters is being restricted to a date range, the backend should support sorting those encounters by date, probably defaulting to ascending order.
2. We could create a specialized endpoint to return the list of forms with the last completed date. More generally, I think an endpoint for forms that could take a "patient" and maybe "visit" would likely be generally useful for more targeted form lists.

**Comment by mogoodrich** [comment]
> This suggests to me two tickets for the backend:
> 
> 1. When the list of encounters is being restricted to a date range, the backend should support sorting those encounters by date, probably defaulting to ascending order.

Agree re: this... though we should first confirm such an endpoint doesn't already exist, it's possible it does and whoever first build this didn't realize it.

**Comment by denniskigen** [comment]
Will you take care of filing those, @ibacher? I'll proceed to merge this in. Thanks for the review.

**Pull request merged**